File: mario.py
import cs50

def main():

    #retrieve the pyramid height from the user
    print("What height of pyramid do you want?")
    height = cs50.get_int()
    #ensure height is not 0
    #ensure height is greater than 1 and less than 23
    if height < 0 or height > 23 or height == 0:
        print("Height needs to be greater than 0 and less than 23.")
    else:
        #determine number of rows
        for i in range(height):
            #determine number of spaces
            for s in range(height - i - 1):
                #print spaces
                print(" ", end="")
            #determine number of hashtags
            for h in range(i+2):
                #print hashtags
                print("#", end="")
            #print new line
            print()

if __name__ == "__main__":
    main()

File: mario.c
#include <stdio.h>
#include <cs50.h>

int main (void)
{
    printf("Height: ");
    int height;

    //retrieve the pyramid height from the user until it is an allowed number
    do
    {
        height = get_int();
        if (height == 0)
            return false;
    }
    while (height < 1 || height > 23);

   //determine the number of rows
    for (int i = 0; i < height ; i++)
    {
        //determine the number spaces
        for (int s = 0; s < height - i - 1; s++)
        {
            //prints the spaces
            printf("%s", " ");
        }
        //determines the number of hashtags
        for (int h = 0; h < i + 2 ; h++)
        {
            //prints the hashtags
            printf("#");
        }
        //prints a new line before going on to the next row
        printf("\n");
    }
}



